lib/rack/tracker.rb

```diff
@@ -64,28 +64,23 @@ def inject(env, response)
     end
 
     class HandlerSet
-      class Handler
-        def initialize(name, options)
-          @name = name
-          @options = options
-        end
-
+      Handler = Struct.new(:name, :configuration) do
         def init(env)
-          @name.new(env, @options)
+          name.new(env, configuration)
         end
       end
 
       def initialize(&block)
         @handlers = []
-        self.instance_exec(&block) if block_given?
+        instance_exec(&block) if block_given?
       end
 
-      def handler(name, opts = {}, &block)
-        @handlers << Handler.new(Rack::Tracker::HandlerDelegator.handler(name), opts)
+      def handler(name, configuration = {}, &block)
+        @handlers << Handler.new(Rack::Tracker::HandlerDelegator.handler(name), configuration)
```

Comment:
I really don't like this... :/ 

but we need "something" (which is atm the _handler_ struct) to postpone the initialization of the _real handler_, to give it the env and configuration options when the result of the handler is injected into the response.